src/nbl/asset/interchange/CIESProfileLoader.cpp

```diff
@@ -0,0 +1,226 @@
+     int hSize = getInt("hSize truncated");
+
+     int type_ = getInt("type truncated");
+     if (error)
+         return false;
+     if (type_ <= 0 || type_ > 3) {
+         errorMsg = "unrecognized type";
+         return false;
+     }
+     CIESProfile::PhotometricType type =
+         static_cast<CIESProfile::PhotometricType>(type_);
+     assert(type == CIESProfile::PhotometricType::TYPE_C &&
+         "Only type C is supported for now");
+
+     int unitsType = getInt("unitsType truncated");
+     double width = getDouble("width truncated"),
+         length = getDouble("length truncated"),
+         height = getDouble("height truncated");
+     double ballastFactor = getDouble("ballastFactor truncated");
+     double reserved = getDouble("reserved truncated");
+     double inputWatts = getDouble("inputWatts truncated");
+     if (error)
+         return false;
+
+     result = CIESProfile(type, hSize, vSize);
+     auto& vAngles = result.getVertAngles();
+     for (int i = 0; i < vSize; i++) {
+         vAngles[i] = getDouble("vertical angle truncated");
+         if (i != 0 && vAngles[i - 1] > vAngles[i])
+             return false; // Angles should be sorted
+     }
+     assert((vAngles[0] == 0.0 || vAngles[0] == 90.0) &&
+         "First angle must be 0 or 90 in type C");
+     assert((vAngles[vSize - 1] == 90.0 || vAngles[vSize - 1] == 180.0) &&
+         "Last angle must be 90 or 180 in type C");
+
+     auto& hAngles = result.getHoriAngles();
+     for (int i = 0; i < hSize; i++) {
+         hAngles[i] = getDouble("horizontal angle truncated");
+         if (i != 0 && hAngles[i - 1] > hAngles[i])
+             return false; // Angles should be sorted
+     }
+     assert((hAngles[0] == 0.0) && "First angle must be 0 in type C");
+     assert((hAngles[hSize - 1] == 0.0 || hAngles[hSize - 1] == 90.0 ||
+         hAngles[hSize - 1] == 180.0 || hAngles[hSize - 1] == 360.0) &&
+         "Last angle must be 0, 90, 180 or 360 in type C");
+
+     double factor = ballastFactor * candelaMultiplier;
+     for (int i = 0; i < hSize; i++) {
+         for (int j = 0; j < vSize; j++) {
+             result.setValue(i, j, factor * getDouble("intensity value truncated"));
+         }
+     }
+
+     if (hAngles.back() == 180.0) {
+         for (int i = (int)hAngles.size() - 2; i >= 0; i--) {
+             result.addHoriAngle(360.0 - hAngles[i]);
+             for (int j = 0; j < vSize; j++)
+                 result.setValue(result.getHoriSize() - 1, j, result.getValue(i, j));
+         }
+     }
+
+     return !error;
+ }
+
+
+class CIESProfileMetadata final : public asset::IAssetMetadata {
+public:
+  CIESProfileMetadata(double maxIntensity)
+      : IAssetMetadata(), maxIntensity(maxIntensity) {}
+
+  _NBL_STATIC_INLINE_CONSTEXPR const char *LoaderName = "CIESProfileLoader";
+  const char *getLoaderName() const override { return LoaderName; }
+
+  double getMaxIntensity() const { return maxIntensity; }
+
+private:
+  double maxIntensity;
```

Comment:
we also need the integral over the sphere, and the "last angle" (90 vs 180 tells me if the profile is full sphere or hemisphere, then we know whether to use octahedral or diamond mapping)

Reply:
I still want the `lastAngle` info, but as a boolean `bool fullSphere; // or only a hemisphere?`